ui/src/shared/components/QueryTabList.tsx

```diff
@@ -0,0 +1,55 @@
+import React, {PureComponent} from 'react'
+import QueryMakerTab from 'src/data_explorer/components/QueryMakerTab'
+import buildInfluxQLQuery from 'src/utils/influxql'
+import {QueryConfig, TimeRange} from 'src/types/query'
+
+interface Props {
+  queries: QueryConfig[]
+  onAddQuery: () => void
+  onDeleteQuery: (index: number) => void
+  activeQueryIndex: number
+  setActiveQueryIndex: (index: number) => void
+  timeRange: TimeRange
+}
+
+export default class QueryTabList extends PureComponent<Props> {
+  public render() {
+    const {
+      queries,
+      onAddQuery,
+      onDeleteQuery,
+      activeQueryIndex,
+      setActiveQueryIndex,
+    } = this.props
+
+    return (
+      <div className="query-maker--tabs">
+        {queries.map((q, i) => (
+          <QueryMakerTab
+            key={i}
+            isActive={i === activeQueryIndex}
+            query={q}
+            onSelect={setActiveQueryIndex}
+            onDelete={onDeleteQuery}
+            queryTabText={this.queryTabText(i, q)}
+            queryIndex={i}
+          />
+        ))}
+        <div
+          className="query-maker--new btn btn-sm btn-primary"
+          onClick={onAddQuery}
+        >
+          <span className="icon plus" />
+        </div>
+      </div>
+    )
+  }
+
+  private queryTabText = (i: number, query: QueryConfig): string => {
```

Comment:
i believe sometimes `rawText` can be `null`... this didn't give any errors? if not, don't worry about it -- it'll surface eventually organically.